core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala

```diff
@@ -368,11 +338,26 @@ abstract class QuorumTestHarness extends Logging {
     // Setting the configuration to the same value set on the brokers via TestUtils to keep KRaft based and Zk based controller configs are consistent.
     props.setProperty(ServerLogConfigs.LOG_DELETE_DELAY_MS_CONFIG, "1000")
     val config = new KafkaConfig(props)
+
+    val formatter = new Formatter().
+      setClusterId(Uuid.randomUuid().toString).
+      setNodeId(nodeId)
+    formatter.addDirectory(metadataDir.getAbsolutePath)
+    formatter.setReleaseVersion(metadataVersion)
+    formatter.setUnstableFeatureVersionsEnabled(true)
+    formatter.setControllerListenerName(config.controllerListenerNames.head)
+    formatter.setMetadataLogDirectory(config.metadataLogDir)
+    addFormatterSettings(formatter)
+    //formatter.setFeatureLevel(Features.TRANSACTION_VERSION.featureName,
+    //  Features.TRANSACTION_VERSION.defaultValue(metadataVersion))
```

Comment:
Uncommented code? Let's remove this if it is not needed.

Reply:
removed